ibu-imager/ops/ops.go

```diff
@@ -67,3 +78,78 @@ func (o *ops) RunBashInHostNamespace(command string, args ...string) (string, er
 	args = append([]string{command}, args...)
 	return o.RunInHostNamespace("bash", "-c", strings.Join(args, " "))
 }
+
+// WaitForEtcd waits for the availability of the etcd server at the given healthz endpoint.
+// It continuously checks the healthz endpoint to determine the server's availability.
+// If the etcd server becomes accessible, it returns nil, indicating success. Otherwise, it returns an error.
+func (o *ops) WaitForEtcd(healthzEndpoint string) error {
+	// Use the default HTTP client
+	client := http.DefaultClient
+
+	for {
+		// Send an HTTP GET request to the healthz endpoint
+		resp, err := client.Get(healthzEndpoint)
+		if err != nil {
+			// If there was an error, wait for a short interval before retrying
+			time.Sleep(1 * time.Second)
+			o.log.Info(err)
+			continue
+		}
+
+		// Ensure the response body is closed after reading
+		defer resp.Body.Close()
+
+		// If the server responds with HTTP status 200 (OK), it's up and running
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		// Wait for a short interval before retrying
+		time.Sleep(1 * time.Second)
+	}
+}
```

Comment:
* This is a rather simple, clear function that doesn't do anything worth documenting IMO
* Use `http.Get` because it automatically uses the default client
* Logging only err and not status code seems inconsistent to me. Also the log should contain some context to make the error less scary

```suggestion
func (o *ops) WaitForEtcd(healthzEndpoint string) error {
	for {
		resp, err := http.Get(healthzEndpoint)
		if err != nil {
			o.log.Infof("Waiting for etcd: %s", err)
			time.Sleep(1 * time.Second)
			continue
		}
		defer resp.Body.Close()
		
		if resp.StatusCode != http.StatusOK {
			o.log.Infof("Waiting for etcd, status: %d", resp.StatusCode)
			time.sleep(1 * time.Second)
			continue
		}
		
		return nil
	}
}
```